pkg/daemon/update.go

```diff
@@ -195,6 +198,29 @@ func (dn *Daemon) writeUnits(units []ignv2_2types.Unit) error {
 			return fmt.Errorf("Failed to write systemd unit %q: %v", u.Name, err)
 		}
 
+		// if the unit doesn't note if it should be enabled or disabled then
+		// skip all linking.
+		// if the unit should be enabled, then enable it.
+		// otherwise the unit is disabled. run disableUnit to ensure the unit is
+		// disabled. even if the unit wasn't previously enabled the result will
+		// be fine as disableUnit is idempotent.
+		// Note: we have to check for legacy unit.Enable. if it is true set the
+		// unit.Enabled to true
+		if u.Enable == true {
```

Comment:
I think these cases should be flipped. it seems better to respect the new field over the deprecated one if both are there for some reason. 

otoh, if this is the way that ignition does it (I didn't check) then we should keep it this way to be as in line with the existing ignition semantics as possible.

Reply:
[Ignition follows this path](https://github.com/coreos/ignition/blob/master/internal/exec/stages/files/units.go#L32-L59):
- If `unit.Enable` is `true` then enable the unit and warn about old interface. continue logic
- if `unit.Enabled` is not `nil` and `true` then enable the unit and continue logic
- if `unit.Enabled` is not `nil` and `false` then disable the unit and continue logic
- if `unit.Enabled` is `nil` continue logic

Reply:
To match that I'll change the `else if` to a second `if` block.